scripts/prove_security.py

```
@@ -0,0 +1,504 @@
+                          "in .github/workflows/security-audit.yml")
+
+    content = dockerfile.read_text(encoding="utf-8")
+    issues: list[str] = []
+
+    # Check for non-root USER directive
+    user_lines = [l for l in content.splitlines() if l.strip().startswith("USER")]
+    if not user_lines or user_lines[-1].strip() == "USER root":
+        issues.append("Container runs as root — use USER directive to drop privileges")
+
+    # Check for pinned base image tags
+    from_lines = [l for l in content.splitlines() if l.strip().startswith("FROM")]
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)